### PR TITLE
TST: reuse existing tests for GPU testing

### PR DESCRIPTION
**Pull request by UranusSeven**
## What do these changes do?

Setup a testing mechanism for letting current test cases run on GPU. For test cases that can run both on CPU and GPU,

1. always use the `setup_gpu` fixture since `setup_gpu` still works even there's no cuda device available.

2. mark test case with annotation `@support_cuda` so that it will be picked up when running pytest -m cuda, a.k.a. our GPU test suite.

3. in the test case, use the bool fixture `gpu` to init dataframes, series, and tensors. `gpu` is, by default, set to `False` so that test cases in CPU test suites still run on CPU. Our GPU test suite, however, runs with `pytest -m cuda --gpu`, where `gpu` is set to `True`, making the test cases run on GPU.

## Related issue number

#369 

## Check code requirements

- [x] tests added / passed (if needed)
- [x] Ensure all linting tests pass

### CODE REVIEW

File: python/xorbits/conftest.py [hunk 1]
```diff
@@ -22,6 +22,18 @@
...
 
+def pytest_addoption(parser):
+    parser.addoption(
+        "--gpu", action="store_true", default=False, help="run test cases on GPU."
+    )
+
+
+@pytest.fixture
+def gpu(request):
+    print(f"GPU: {request.config.option.gpu}")
```
Comment (qianduoduo0904) [inline python/xorbits/conftest.py:33]: Remove the print function.

**Comment by qianduoduo0904** [comment]
Seems that I didn't see `setup_gpu` included in this PR?

**Comment by UranusSeven** [comment]
> Seems that I didn't see `setup_gpu` included in this PR?

`setup_gpu` is an existing fixture.

**Review by qianduoduo0904 — APPROVED** [review]
LGTM

**Review by aresnow1 — APPROVED** [review]
LGTM

**Pull request merged**